src/DataflowPass.cpp

```diff
@@ -168,79 +205,132 @@ void DataflowPass::transfer(Instruction *instruction,
       bool loopBroken = false;
       for (auto Pair : this->memoryFunctions) {
         if (fnName == Pair.first) {
-          logout("not calling onalloc for "
-                 << arg << " of fnName " << fnName << " and inst "
-                 << *instruction << " and just returning now") return;
+          logout("not calling onalloc for " << arg << " of fnName " << fnName
+                                            << " and inst " << *instruction
+                                            << " and just returning now");
+          return;
 
         } else if (fnName == Pair.second) {
-          MethodsSet *methods =
-              inputProgramPoint.getPVRef(arg, false)->getMethodsSetRef();
-          logout("GETMAIN-5 ON '"
-                 << arg
-                 << "'") this->onDeallocationFunctionCall(methods, fnName);
+          MethodsSet *methods = pv->getMethodsSetRef();
+          logout("GETMAIN-5 ON '" << arg << "'");
+          this->onDeallocationFunctionCall(methods, fnName);
           loopBroken = true;
           break;
         }
       }
 
-      if (loopBroken) {
+      if (loopBroken || fnName == "llvm.dbg.declare") {
```

Comment:
This extra condition definitely needs a comment to justify it. I think this might be debugging code, and if so it should also be guarded by a debug flag.

Reply:
It's not debugging code; it's just that llvm.dbg.declare functions get lumped in as function calls along with function calls we make in the C code. llvm.dbg.declare are calls generated to declare debug info within the IR. 

At this point, we're looking for functions with annotations, and llvm.dbg.declare will not be a function with annotations, so we'll not check it. I'll add a comment noting this.